packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -40,6 +40,42 @@ import {
   parseCaipChainId,
 } from '@metamask/utils';
 
+/**
+ * A list of stateful component types.
+ */
+const STATEFUL_COMPONENT_TYPES = [
+  'Input',
+  'Dropdown',
+  'RadioGroup',
+  'FileInput',
+  'Checkbox',
+  'Selector',
+  'AssetSelector',
+  'AddressInput',
+] as const;
+
+/**
+ * Type for stateful component types.
+ */
+type StatefulComponentType = (typeof STATEFUL_COMPONENT_TYPES)[number];
+
+/**
+ * Check if a component is a stateful component.
+ *
+ * @param component - The component to check.
+ * @param component.type - The type of the component.
+ *
+ * @returns Whether the component is a stateful component.
+ */
+export function isStatefulComponent(component: { type: string }): component is {
+  type: StatefulComponentType;
+  props: Record<string, any>;
```

Comment:
This is not ideal 🤔 Is this required?

Reply:
Can we at least type it to be a union of the stateful component's props?

Reply:
Thought it was required at some point but it isn't anymore apparently, removed